Doppler.Sap/Mappers/BusinessPartner/BusinessPartnerMapper.cs

```diff
@@ -0,0 +1,34 @@
+using Doppler.Sap.Models;
+using System.Collections.Generic;
+
+namespace Doppler.Sap.Mappers.BusinessPartner
+{
+    public abstract class BusinessPartnerMapper
+    {
+        /// <summary>
+        /// Key: PlanType; Value: GroupCode
+        /// </summary>
+        protected abstract Dictionary<int, int> DopplerGroupCodes { get; }
+
+        /// <summary>
+        /// Key: ClientManagerType; Value: GroupCode
+        /// </summary>
+        protected abstract Dictionary<int, int> ClientManagerGroupCodes { get; }
+
+        /// <summary>
+        /// Key: PlanType; Value: GroupCode
+        /// </summary>
+        protected abstract Dictionary<int, int> RelayGroupCodes { get; }
+
+        protected int MapGroupCode(DopplerUserDto dopplerUser)
+        {
+            var groupCode = (!dopplerUser.IsClientManager && !dopplerUser.IsFromRelay) ?
+                            DopplerGroupCodes.TryGetValue(dopplerUser.PlanType.Value, out var dopplerGroupCode) ? dopplerGroupCode : 0 :
```

Comment:
(dopplerUser.PlanType.Value is int`?`, it can trigger an exception, why PlanType is int`?` and not only int?

Reply:
Let me check

Reply:
I added the check by HasValue becuase from Doppler the PlanType is ```int?```